tests/core/job_test.py

```diff
@@ -272,14 +272,48 @@ def test_restore_state_sets_job_runs(self):
             calls = [mock.call(mock_runs[i]) for i in range(0, len(mock_runs))]
             self.job.watch.assert_has_calls(calls)
 
+    def test_create_and_schedule_runs_specific_time(self):
```

Comment:
I think this whole test case class belongs in job_scheduler_test.py? If you feel like moving it.